apps/meteor/app/authentication/server/startup/index.js

```diff
@@ -199,77 +199,76 @@ const validateEmailDomain = (user) => {
+			if (options.profile) {
+				if (options.profile.name) {
+					user.name = options.profile.name;
+				} else if (options.profile.firstName) {
+					// LinkedIn format
+					user.name = getLinkedInName(options.profile);
+				}
 			}
 		}
-	}
 
-	if (user.services) {
-		const verified = settings.get('Accounts_Verify_Email_For_External_Accounts');
+		if (user.services) {
+			const verified = settings.get('Accounts_Verify_Email_For_External_Accounts');
 
-		for (const service of Object.values(user.services)) {
-			if (!user.name) {
-				user.name = service.name || service.username;
-			}
+			for (const service of Object.values(user.services)) {
+				if (!user.name) {
+					user.name = service.name || service.username;
+				}
 
-			if (!user.emails && service.email) {
-				user.emails = [
-					{
-						address: service.email,
-						verified,
-					},
-				];
+				if (!user.emails && service.email) {
+					user.emails = [
+						{
+							address: service.email,
+							verified,
+						},
+					];
+				}
 			}
 		}
-	}
-
-	if (!options.skipAdminEmail && !user.active) {
-		const destinations = [];
-		const usersInRole = await Roles.findUsersInRole('admin');
-		await usersInRole.forEach((adminUser) => {
-			if (Array.isArray(adminUser.emails)) {
-				adminUser.emails.forEach((email) => {
-					destinations.push(`${adminUser.name}<${email.address}>`);
-				});
-			}
-		});
 
-		const email = {
-			to: destinations,
-			from: settings.get('From_Email'),
-			subject: Accounts.emailTemplates.userToActivate.subject(),
-			html: Accounts.emailTemplates.userToActivate.html({
-				...options,
-				name: options.name || options.profile?.name,
-				email: options.email || user.emails[0].address,
-			}),
-		};
+		if (!options.skipAdminEmail && !user.active) {
+			const usersInRole = await Roles.findUsersInRole('admin').toArray();
+			const destinations = usersInRole.flatMap(adminUser =>
+				adminUser.emails?.map(email => `${adminUser.name}<${email.address}>`) ?? []
+			);
+
+			const email = {
+				to: destinations,
+				from: settings.get('From_Email'),
+				subject: Accounts.emailTemplates.userToActivate.subject(),
+				html: Accounts.emailTemplates.userToActivate.html({
+					...options,
+					name: options.name || options.profile?.name,
+					email: options.email || user.emails[0].address,
+				}),
+			};
+
+			await Mailer.send(email);
+		}
 
-		await Mailer.send(email);
-	}
+		if (!options.skipOnCreateUserCallback) {
+			await callbacks.run('onCreateUser', options, user);
+		}
 
-	if (!options.skipOnCreateUserCallback) {
-		await callbacks.run('onCreateUser', options, user);
-	}
+		if (!options.skipEmailValidation && !validateEmailDomain(user)) {
+			throw new Meteor.Error(403, 'User validation failed');
+		}
 
-	if (!options.skipEmailValidation && !validateEmailDomain(user)) {
-		throw new Meteor.Error(403, 'User validation failed');
+		return user;
+	} catch (error) {
+		throw new Meteor.Error('error-on-create-user', error.message);
 	}
```

Comment:
we are expecting that this error is gonna be thrown, with this try/catch we no longer prevent user creation in case of some unexpected error